fhttp/http.go

```diff
@@ -750,9 +752,81 @@ var (
 	EchoRequests int64
 )
 
+// Format: status=503 for 100% 503s
+// status=503:20,404:10 for 20% 503s, 10% 404s, 70% 200s
+func parseStatus(status string) int {
+	lst := strings.Split(status, ",")
+	log.Debugf("Parsing status %s -> %v", status, lst)
+	// Simple non probabilistic status case:
+	if len(lst) == 1 && !strings.ContainsRune(status, ':') {
+		s, err := strconv.Atoi(status)
+		if err != nil {
+			log.Warnf("Bad input status %v, not a number nor coma and colon separated % list", status)
+			return http.StatusBadRequest
+		}
+		log.Debugf("Parsed status %s -> %d", status, s)
+		return s
+	}
+	probTable := make([]int, 100)
+	for i := len(probTable) - 1; i >= 0; i-- {
+		probTable[i] = http.StatusOK
+	}
+	lastPercent := 0
+	for _, entry := range lst {
+		l2 := strings.Split(entry, ":")
+		if len(l2) != 2 {
+			log.Warnf("Should have exactly 1 : in status list %s -> %v", status, entry)
+			return http.StatusBadRequest
+		}
+		s, err := strconv.Atoi(l2[0])
+		if err != nil {
+			log.Warnf("Bad input status %v -> %v, not a number before colon", status, l2[0])
+			return http.StatusBadRequest
+		}
+		percStr := l2[1]
+		p, err := strconv.Atoi(percStr)
+		if err != nil || p < 0 || p > 100 {
+			log.Warnf("Percentage is not a [0 - 100] number in %v -> %v : %v %d", status, percStr, err, p)
+			return http.StatusBadRequest
+		}
+		if lastPercent+p > 100 {
```

Comment:
[thinking out loud] this seems fine. i wonder if it might be nice to autoscale to 100 down-the-road. like if say:

`?status:404:95,302:15,200:90`

we would adjust down to something like:

```
404 -> 48
302 -> 7
200 -> 45
```

That might be a bit surprising, but might help correct for silly math blunders too...